File: python/cookbook/recipe_file.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-

import os
from cookbook.util.logging import LoggerFactory

logger = LoggerFactory.getLogger(__name__)

def title():

    logger.debug('# File')

def cook():

    name = os.path.abspath('cooker.py')

    # 基本情報

    logger.debug("dir=%s" % (os.path.dirname(name)))
    logger.debug("file=%s" % (os.path.basename(name)))
    logger.debug("size=%d" % (os.path.getsize(name)))

    # 行数カウント

    lc = 0
    with open(name, 'r') as file:
        lc = sum(1 for line in file)
    logger.debug("lc=%d" % (lc))

    lc = 0
    with open(name, 'r') as file:
        lc = len(file.readlines())
    logger.debug("lc=%d" % (lc))

    lc = 0
    with open(name, 'r') as file:
        for line in file:
            lc += 1
    logger.debug("lc=%d" % (lc))

if __name__ == '__main__':

    title()
    cook()

File: ruby/cookbook/recipe_file.rb
#!/usr/bin/env ruby

module Recipe

    require './cookbook/util/logging'

    class << self

        @@logger = LoggerFactory.getLogger File.basename __FILE__

        def title

            @@logger.debug '# File'

        end

        def cook

            name =  File.absolute_path('cooker.rb')

            # 基本情報

            @@logger.debug "dir=#{File.dirname(name)}"
            @@logger.debug "file=#{File.basename(name)}"
            @@logger.debug "size=#{File.size(name)}"

            # 行数カウント

            lc = File.readlines(name).size
            @@logger.debug "lc=#{lc}"

            lc = File.foreach(name).count
            @@logger.debug "lc=#{lc}"

            lc = 0
            open(name, 'r') do |file|
                file.each_line do |line|
                    lc += 1
                end
            end
            @@logger.debug "lc=#{lc}"

        end

    end

end

if __FILE__ == $0

    Recipe.title
    Recipe.cook

end
